NitroxServer-Subnautica/Serialization/ResourceAssetsParser.cs

```diff
@@ -58,17 +82,457 @@ public static ResourceAssets Parse()
+        }
+
+        private static AssetFileInfoEx FindComponentOfType(Type type, AssetsFileInstance assetsFileInstance, ulong gameObjectPathId)
+        {
+            AssetsFileTable assetsTable = assetsFileInstance.table;
+            AssetsFile assetsFile = assetsFileInstance.file;
+            AssetsFileReader afr = assetsFile.reader;
+
+            ulong prevPos = assetsFile.reader.Position;
+            AssetFileInfoEx gameObjectAFI = assetsTable.getAssetInfo(gameObjectPathId);
+            if (gameObjectAFI.curFileType != 1)
+            {
+                Log.Info("An incorrect pathId was passed to FindComponentOfType please check if it points to a proper gameObject: " + gameObjectPathId);
+                return null;
+            }
+            afr.Position = gameObjectAFI.absoluteFilePos;
+            afr.Align();
+
+            uint componentCount = afr.ReadUInt32();
+            AssetFileInfoEx componentAFI;
+
+            for (int i = 0; i < componentCount; i++)
+            {
+                uint fileId = (uint)afr.ReadInt32();
+                ulong pathId = (uint)afr.ReadInt64();
+                if (fileId == 0)
+                {
+                    componentAFI = assetsTable.getAssetInfo(pathId);
+                    if (componentAFI.curFileType == 114)
+                    {
+                        ulong curPos = afr.Position;
+                        afr.Position = componentAFI.absoluteFilePos;
+                        afr.Align();
+                        afr.Position += 16;
+
+                        int scriptFileId = afr.ReadInt32();
+                        long scriptPathId = afr.ReadInt64();
+
+                        if (matchingMonoscripts.Contains((ulong)scriptPathId))
+                        {
+                            return componentAFI;
+                        }
+
+                        AssetsFileInstance monoscriptDep = assetsFileInstance.dependencies[scriptFileId - 1];
+                        AssetFileInfoEx monoscriptAFI = monoscriptDep.table.getAssetInfo((ulong)scriptPathId);
+                        if (monoscriptAFI.curFileType == 115)
+                        {
+                            monoscriptDep.file.reader.Position = monoscriptAFI.absoluteFilePos;
+                            monoscriptDep.file.reader.Align();
+
+                            string monoscriptName = monoscriptDep.file.reader.ReadCountStringInt32();
+
+                            if (monoscriptName == type.Name)
+                            {
+                                afr.Position = prevPos;
+                                return componentAFI;
+                            }
+                        }
+                        afr.Position = curPos;
+                    }
+                }
+            }
+
+            afr.Position = prevPos;
+
+            return null;
+        }
+
+        public static List<UwePrefab> GetPrefabFromPlaceholder(string classId)
```

Comment:
Wasn't this method in the class above?  Shouldn't this class just parse all of the data then have it passed to the consumers later?